README.md

```diff
@@ -1,40 +1,50 @@
 # cd4pe_deployments
 
-//TODO:
-- get information about the environment deployments will be running in (available vars, restrictions, secrets manaagement, etc)
-- figure out the development section, how will others develop new policies and test them?
-
 #### Table of Contents
 
 1. [Description](#description)
-2. [Usage - Configuration options and additional functionality](#usage)
+2. [Usage](#usage)
+    * [Build your own policy](#build-your-own-policy)
+    * [Error handling](#error-handling)
 3. [Reference](#reference)
-4. [Development - Guide for contributing to the module](#development)
+4. [Development](#development)
 
 ## Description
 
-This module provides a set of tools, via CD4PE, for creating your own custom CD4PE deployment policies. It is also a place to find a set of generic policies.
+This module provides a set of tools, via CD4PE, for creating your own custom CD4PE deployment policies as well as our 4 built-in policies: [Feature branch](plans/feature_branch.pp), [Temporary branch](plans/rolling.pp), [Direct](plans/direct.pp), and [Eventual consistency](plans/eventual_consistency.pp).
 
 ## Usage
 
+### Build your own policy
+
 1. Get started with a custom deployment policy by creating a [Puppet Plan] in the `plans/` directory.
-2. You'll have access to all of the functions listed in our [REFERENCE.md](REFERENCE.md) to perform deployment operations (e.g. pinning nodes to an environment group or getting information about an environment group)
+2. You'll have access to all of the functions listed in our [REFERENCE.md] to perform deployment operations (e.g. pinning nodes to an environment group or getting information about an environment group)
 3. Deployment policies will run inside a CD4PE context with certain environment variables available.
 
-```
-myplan.yaml
+    * `BRANCH`: the name of the branch you are deploying _from_; your pipeline branch
+    * `COMMIT`: HEAD commit SHA of the branch you are deploying from
+    * `NODE_GROUP_ID`: alpha-numberic ID of the node group you are deploying _to_
+    * `REPO_TARGET_BRANCH`: name of the source control branch that represents the target environment
+    * `ENVIRONMENT_PREFIX`: if you are deploying to a prefixed environment, this is that prefix
+    * `REPO_TYPE`: will either be CONTROL_REPO or MDOULE
+    * `DEPLOYMENT_DOMAIN`: domain of the workspace that initiates the deployment
+    * `DEPLOYMENT_ID`: unique ID of the deployment
+    * `DEPLOYMENT_TOKEN`: the auth token that authenticates deployment tasks against CD4PE
+    * `DEPLOYMENT_OWNER`: user that intitiates the deployment
```

Comment:
Theres more then just this list. https://github.com/puppetlabs/PipelinesInfra-common/blob/master/impl/src/main/java/com/puppet/cdpe/deployment/DeploymentTask.java#L537